File: 6_single_inheritance.py
class Shape:
    width = 0.0
    height = 0.0

    def setWidth(self, w):
        self.width = w

    def setHeight(self, h):
        self.height = h


class Rectangle(Shape):
    def getArea(self):
        return self.width * self.height


rect = Rectangle()
rect.setWidth(4.5)
rect.setHeight(2)
print("Total area:", rect.getArea())

File: 6_single_inheritance.cpp
#include <bits/stdc++.h>
using namespace std;

class Shape
{
protected:
    double width, height;

public:
    void setWidth(double w)
    {
        width = w;
    }
    void setHeight(double h)
    {
        height = h;
    }
};

class Rectangle : public Shape
{
public:
    double getArea()
    {
        return width * height;
    }
};

int main()
{
    Rectangle rect;

    rect.setWidth(4.5);
    rect.setHeight(2);
    cout << "Total area: " << rect.getArea() << endl;

    return 0;
}